.github/workflows/node_modules-publish.yml

```diff
@@ -11,6 +11,8 @@ jobs:
     timeout-minutes: 5
     env:
       GIT_DIR: git-dir.git
+      HEAD_SHA: ${{ github.event.workflow_run.pull_requests[0].head.sha }}
+      BASE_SHA: ${{ github.event.workflow_run.pull_requests[0].base.sha }}
```

Comment:
`github.event.workflow_run.base_sha` isn't a thing then?

Reply:
No.  The head sha gets promoted but the base sha doesn't, unfortunately.  And then I use the head sha for consistency.  Although I'm starting to wonder if that gives us the correct result when people push out-of-date branches.  I'm going to need to do a bit more research here unfortunately :(

Reply:
Result of research is here:

https://gist.github.com/allisonkarlitskaya/32a48ef569d58d43c44c8d0bbee0e144

The long story short is that we don't have access to the PR base SHA from `workflow_run` events.  We can make up for it by using `github.sha` which is the current state of the branch, which is close enough when combined with the tricks I'm using in #15854.

Meanwhile, my approach here is also bogus: for some reason, the `pull_requests[]` array is only populated for pull requests opened from a branch of the same repository, which is kinda useless for us.

It also explains why no comments were being posted: we iterate over an empty array.

I think I'm going to have to send the PR number along with the artifacts from the build workflow.  We can be sure that workflow hasn't been modified by using the `git log` tricks.

This stuff sure isn't easy to get right...